yt/frontends/enzo/io.py

```diff
@@ -75,12 +78,25 @@ def _read_particle_fields(self, chunks, ptf, selector):
                     continue
                 ds = f.get("/Grid%08i" % g.id)
                 for ptype, field_list in sorted(ptf.items()):
-                    if ptype != "io":
+                    if ptype == "io":
+                        if g.NumberOfParticles == 0:
+                            continue
+                        pds = ds
+                    elif ptype == "DarkMatter":
                         if g.NumberOfActiveParticles[ptype] == 0:
                             continue
-                        pds = ds.get(f"Particles/{ptype}")
-                    else:
                         pds = ds
+                    else:
+                        if g.NumberOfActiveParticles[ptype] == 0:
+                            continue
+                        for pname in ["Active Particles", "Particles"]:
+                            pds = ds.get(f"{pname}/{ptype}")
+                            if pds is not None:
+                                break
+                        if pds is None:
```

Comment:
```suggestion
                        else:
```

Reply:
It's not obvious to me that `else` is the right approach here. The 'if' statement is checking if 'pds' is 'None'

Reply:
but that should only ever happen in case the `break` statement above isn't hit, correct ?
The `else` in `for` loops can be read as `if no break`. I've thought about this for a a minute and I really can't see any case where it's not equivalent, because the for loop is _always_ going to run (given the iterable is hard-coded as non-empty), so I can't see how `pds` could end up as `None` in any other scenario than "the break wasn't hit"